src/com/xilinx/rapidwright/util/VivadoTools.java

```diff
@@ -120,19 +120,58 @@ public static List<String> runTcl(Path outputLog, Path tclScript, boolean verbos
      * @return ReportRouteStatusResult object.
      */
     public static ReportRouteStatusResult reportRouteStatus(Design design) {
+        final Path dcp = writeCheckpoint(design);
+        boolean encrypted = !design.getNetlist().getEncryptedCells().isEmpty();
+        ReportRouteStatusResult rrs = reportRouteStatus(dcp, dcp.getParent(), encrypted);
+
+        FileTools.deleteFolder(dcp.getParent().toString());
+
+        return rrs;
+    }
+
+    private static Path writeCheckpoint(Design design) {
         final Path workdir = FileSystems.getDefault()
                 .getPath("vivadoToolsWorkdir" + FileTools.getUniqueProcessAndHostID());
         File workdirHandle = new File(workdir.toString());
         workdirHandle.mkdirs();
         final Path dcp = workdir.resolve("checkpoint.dcp");
         design.writeCheckpoint(dcp);
+        return dcp;
+    }
 
-        boolean encrypted = !design.getNetlist().getEncryptedCells().isEmpty();
-        ReportRouteStatusResult rrs = reportRouteStatus(dcp, workdir, encrypted);
-
-        FileTools.deleteFolder(workdir.toString());
+    /**
+     * Run Vivado's `write_bitstream` on the provided DCP file to generate a bit
+     * file at the specified location.
+     * 
+     * @param dcp            The DCP file from which to generate a bitstream.
+     * @param bitFile        The location of the bit file to generate
+     * @param hasEncryptedIP Flag indicating if the provided DCP contains encrypted
+     *                       IP and was written by RapidWright such that it needs to
+     *                       be loaded with a Tcl script.
+     * @return The output of Vivado as a list of Strings
+     */
+    public static List<String> createBitstream(Path dcp, Path bitFile, boolean hasEncryptedIP) {
```

Comment:
Would calling this method `writeBitstream` be a better alignment with Vivado terminology?

Reply:
Thanks, I updated it.